src/middlewared/middlewared/plugins/nfs.py

```diff
@@ -448,5 +451,40 @@ def validate_hosts_and_networks(self, other_shares, data, schema_name, verrors,
             break
 
 
+class NFSFSAttachmentDelegate(FSAttachmentDelegate):
+    name = 'nfs'
+    title = 'NFS Share'
+
+    async def query(self, path, enabled):
+        results = []
+        for nfs in await self.middleware.call('sharing.nfs.query', [['enabled', '=', enabled]]):
+            if any(is_child(nfs_path, path) for nfs_path in nfs['paths']):
+                results.append(nfs)
+
+        return results
+
+    async def get_attachment_name(self, attachment):
+        return ', '.join(attachment['paths'])
+
+    # NFS share can only contain paths from single dataset, that's why we can delete/disable entire share
+    # if even one path matches
+    async def delete(self, attachments):
+        for attachment in attachments:
+            await self.middleware.call(
+                'datastore.delete', 'sharing.nfs_share_path', [['share_id', '=', attachment['id']]]
+            )
```

Comment:
Do we really need to delete nfs_share_path first?
I always thought django ORM would delete it because its a FK with default action to delete related entries.